# Film Funding Factors Analysis

**Abeer Radhi**

## Technical Report

### Introduction
Film funding decisions are made before financial outcomes are known. This project examines how factors available before release, including genre, production budget, previous lead experience, runtime, and available prior creative team history, relates to a film's **reported return**.

The analysis combines IMDb film, cast, crew, and genre information with TMDB reported budget, revenue, and production-company data. The main analysis covers **2,465 selected films released from 2014 to 2023**.

### Financial Measure
**Reported Return = Reported Revenue minus Production Budget, divided by Production Budget**

This is a simplified reported-return measure. It excludes marketing, distribution, cinema/distributor shares, financing costs, and investor payments. It is therefore **not actual investor profit or a true break-even measure**.

A film is described as **below budget** when its reported revenue is lower than its reported production budget.

### Problem Statement
While film studios consider factors such as genre, budget, and previous lead experience when assessing a film's potential success before funding it, these factors showed weak relationships with reported return across **2,465 selected IMDb and TMDB films released from 2014 to 2023**. In this sample, **34.8% (858 films)** reported revenue below production budget, with an **average shortfall of $11.05 million** among those films. This creates uncertainty around which factors should receive greater consideration before funding a film.

### Objectives
1. Determine the relationship between budget, previous lead experience, genre, and reported return.
2. Identify genres with the lowest rates of films below budget.
3. Determine how the rate of films below budget varies across budget levels.
4. Compare reported return and below budget rates by previous lead experience.
5. Compare reported return across director and writer prior-history groups.
6. **Combined model:** Determine which factors show clearer relationships with reported return when considered together.

### Target Audience
**Primary:** Studio greenlight teams who decide which film projects move forward.

**Secondary:** Investors, financiers, and content producers.

### Interpretation Scope
The project is **descriptive and exploratory**. Correlations, group comparisons, and regression results show relationships within this selected historical sample; they do not establish causation, and the regression was **not tested on new films**.


## Data Sources and Data Dictionary

### Dataset Description
Five raw source files were used. IMDb supplied `name.basics`, `title.ratings`, `title.basics`, and `title.principals`. TMDB financial and film metadata came from the TMDB Movies Dataset 2023 on Kaggle. The raw sources were loaded or queried through SQLite where necessary, filtered, joined using IMDb title identifiers, and transformed into film level analysis data.

The final main cleaned dataset contains **2,465 films and 41 columns** covering **2014 to 2023**. One row represents one film. The separate cleaned cast and crew table contains **37,483 rows and 6 columns** and links films to actors, actresses, directors, and writers through IMDb identifiers.

### Main Data Dictionary

| Column | Description | Type |
| --- | --- | --- |
| `imdb_id` | IMDb title identifier used as join key | String |
| `title` | Film title | String |
| `release_year` | Film release year | Integer |
| `runtime_minutes` | Film runtime in minutes | Numeric |
| `genres` | Comma separated IMDb genre categories | String |
| `budget` | Reported production budget; treated as USD | Numeric |
| `revenue` | Reported film revenue; treated as USD | Numeric |
| `Reported Return` | Reported return measure: (revenue minus budget) divided by budget | Numeric |
| `imdb_rating` | IMDb audience rating; descriptive only, not used as a before funding model predictor | Numeric |
| `imdb_vote_count` | Number of IMDb rating votes; descriptive only, not used as a before funding model predictor | Numeric |
| `original_language` | Original film language from TMDB | String |
| `production_companies` | Comma separated production companies from TMDB | String |
| `popularity` | TMDB popularity measure; retained for reference but not used as a before funding model predictor | Numeric |
| `rec_director` | 1 if at least one credited director had a prior directing film before the current release year | Integer (0/1) |
| `rec_writer` | 1 if at least one credited writer had a prior writing film before the current release year | Integer (0/1) |
| `star_power` | **Legacy technical field name** for Previous Lead Experience: 1 if at least one of the first two credited lead performers had a prior lead film before release | Integer (0/1) |
| `genre_Action` | Genre indicator: 1 when the film is classified as Action, otherwise 0 | Integer (0/1) |
| `genre_Adventure` | Genre indicator: 1 when the film is classified as Adventure, otherwise 0 | Integer (0/1) |
| `genre_Animation` | Genre indicator: 1 when the film is classified as Animation, otherwise 0 | Integer (0/1) |
| `genre_Biography` | Genre indicator: 1 when the film is classified as Biography, otherwise 0 | Integer (0/1) |
| `genre_Comedy` | Genre indicator: 1 when the film is classified as Comedy, otherwise 0 | Integer (0/1) |
| `genre_Crime` | Genre indicator: 1 when the film is classified as Crime, otherwise 0 | Integer (0/1) |
| `genre_Documentary` | Genre indicator: 1 when the film is classified as Documentary, otherwise 0 | Integer (0/1) |
| `genre_Drama` | Genre indicator: 1 when the film is classified as Drama, otherwise 0 | Integer (0/1) |
| `genre_Family` | Genre indicator: 1 when the film is classified as Family, otherwise 0 | Integer (0/1) |
| `genre_Fantasy` | Genre indicator: 1 when the film is classified as Fantasy, otherwise 0 | Integer (0/1) |
| `genre_History` | Genre indicator: 1 when the film is classified as History, otherwise 0 | Integer (0/1) |
| `genre_Horror` | Genre indicator: 1 when the film is classified as Horror, otherwise 0 | Integer (0/1) |
| `genre_Music` | Genre indicator: 1 when the film is classified as Music, otherwise 0 | Integer (0/1) |
| `genre_Musical` | Genre indicator: 1 when the film is classified as Musical, otherwise 0 | Integer (0/1) |
| `genre_Mystery` | Genre indicator: 1 when the film is classified as Mystery, otherwise 0 | Integer (0/1) |
| `genre_News` | Genre indicator: 1 when the film is classified as News, otherwise 0 | Integer (0/1) |
| `genre_Romance` | Genre indicator: 1 when the film is classified as Romance, otherwise 0 | Integer (0/1) |
| `genre_Sci-Fi` | Genre indicator: 1 when the film is classified as Sci Fi, otherwise 0 | Integer (0/1) |
| `genre_Sport` | Genre indicator: 1 when the film is classified as Sport, otherwise 0 | Integer (0/1) |
| `genre_Thriller` | Genre indicator: 1 when the film is classified as Thriller, otherwise 0 | Integer (0/1) |
| `genre_War` | Genre indicator: 1 when the film is classified as War, otherwise 0 | Integer (0/1) |
| `genre_Western` | Genre indicator: 1 when the film is classified as Western, otherwise 0 | Integer (0/1) |
| `historical_genre_roi` | Average Reported Return of films in the same genre(s) released in earlier years only | Numeric |
| `historical_director_roi` | Average prior year Reported Return of the credited director(s), averaged to film level | Numeric |
| `historical_writer_roi` | Average prior year Reported Return of the credited writer(s), averaged to film level | Numeric |

### Cast and Crew Data Dictionary

| Column | Description | Type |
| --- | --- | --- |
| `imdb_id` | IMDb film identifier linking the credit to the main film dataset | String |
| `category` | Credit role: actor, actress, director, or writer | String |
| `ordering` | IMDb principal credit ordering within the title | Integer |
| `nconst` | IMDb person identifier | String |
| `primaryName` | Person's primary IMDb name | String |
| `knownForTitles` | IMDb known for title identifiers | String |

### Key Measures
**Reported Return (technical column: `ROI`)** = `(revenue, budget) / budget`.

**Reported Revenue Below Production Budget** is defined as `Reported Return < 0`. This means the reported revenue is lower than the reported production budget. It does not represent actual financial loss because other costs are not included.

Historical genre, director, and writer measures use only films released in **earlier years**. Same year films are excluded from prior history because the IMDb source used in this project provides release year rather than the exact release order within each year.

### Analysis Factor Scope

The analysis examines factors available before or around the investment and production decision, including budget, genre, runtime, previous lead experience, and available prior history. IMDb rating, vote count, TMDB popularity, and revenue are not included as explanatory factors because they reflect audience response after release or are used to calculate Reported Return.


### Coverage and Missingness Relevant to the Final Presentation
- Final analytical sample: **2,465 unique films**, 2014 to 2023.
- Duplicate final IMDb IDs: **0**.
- Missing runtime: **23 films**.
- Director prior-return history available: **633 films (25.7%)**.
- Writer prior-return history available: **792 films (32.1%)**.
- Director and writer history is **prior history available in this selected dataset, not full career history**.
- Same-year releases are excluded from prior-history calculations.


### Tools Used
- **Python**, Data cleaning, feature engineering, statistical testing, and regression modeling.
- **Tableau**, Primary tool for data analysis and visualization.
- **SQL (SQLite)**, Processing and filtering large IMDb datasets.


## Data Preparation


### Step 1: Load the Source Files

Five raw files were used: four from IMDb (name.basics, title.ratings, title.basics, title.principals) and one from TMDB (Kaggle).

Loading title.principals.tsv.gz in full with `pandas.read_csv` failed with a `MemoryError`, since the file is about 4.4 GB. The cell below is wrapped in `if False:` so it stays as real, readable code but never actually executes, even with "Run All".


In [1]:
# Wrapped in `if False:` on purpose - this attempt caused a MemoryError
# (see Step 1b for the working approach). Kept here as code, not just text,
# but disabled so it can never run again by accident.
if False:
    name_basics = pd.read_csv(RAW_PATH / "name.basics.tsv.gz", sep="\t", low_memory=False)
    title_ratings = pd.read_csv(RAW_PATH / "title.ratings.tsv.gz", sep="\t")
    title_basics = pd.read_csv(RAW_PATH / "title.basics.tsv.gz", sep="\t", low_memory=False)
    title_principals = pd.read_csv(RAW_PATH / "title.principals.tsv.gz", sep="\t", low_memory=False)  # MemoryError here
    tmdb = pd.read_csv(RAW_PATH / "TMDB_movie_dataset_v11.csv", low_memory=False)

    datasets = {
        "name.basics": name_basics,
        "title.ratings": title_ratings,
        "title.basics": title_basics,
        "title.principals": title_principals,
        "TMDB": tmdb
    }

    # Initial inspection for each file
    for name, df in datasets.items():
        print(f"--- {name} ---")
        print(f"Shape: {df.shape[0]:,} rows, {df.shape[1]} columns")
        print(f"Columns: {df.columns.tolist()}")
        print(f"Duplicate rows: {df.duplicated().sum():,}")
        print(f"Missing values per column:\n{df.isnull().sum()}\n")

### Step 2: Load the Source Files into SQLite

To work around the memory error, all five files were loaded into one SQLite database (`raw_data.db`), one table per file, reading each file in chunks so no single file is ever fully loaded into memory at once.

This only needs to run once. `raw_data.db` was used as a local working database during development. The original raw source files and the cleaned intermediate outputs are included with the submission, so rebuilding this optional database is not required to review the submitted analysis. The cell below is wrapped in `if False:` to avoid re running several gigabytes of loading by accident.


In [2]:
# Wrapped in `if False:` - already run once, raw_data.db exists on disk
if False:
    RAW_PATH = Path("datasets/raw")
    DB_PATH = RAW_PATH / "raw_data.db"
    conn = sqlite3.connect(DB_PATH)

    # (file name, separator, table name) for each raw file
    files_to_load = [
        ("name.basics.tsv.gz", "\t", "name_basics"),
        ("title.ratings.tsv.gz", "\t", "title_ratings"),
        ("title.basics.tsv.gz", "\t", "title_basics"),
        ("title.principals.tsv.gz", "\t", "title_principals"),
        ("TMDB_movie_dataset_v11.csv", ",", "tmdb"),
    ]

    chunk_size = 500_000  # rows per chunk, small enough to avoid the MemoryError above

    for file_name, sep, table_name in files_to_load:
        print(f"Loading {file_name} into table '{table_name}'...")
        first_chunk = True
        for chunk in pd.read_csv(RAW_PATH / file_name, sep=sep, chunksize=chunk_size, low_memory=False):
            chunk.to_sql(table_name, conn, if_exists="replace" if first_chunk else "append", index=False)
            first_chunk = False
        count = pd.read_sql(f"SELECT COUNT(*) AS rows FROM {table_name}", conn)
        print(f"  -> {count['rows'][0]:,} rows loaded\n")

    conn.close()
    print("All files loaded into raw_data.db")

### Step 3: Explore and Filter the Source Tables

Running the filtering queries through pandas (`pd.read_sql`) was very slow in this environment, likely due to the size of `raw_data.db` and slow disk access. All exploration and filtering was instead done directly in **DB Browser for SQLite**.

1. **name_basics**: checked row count, a sample of the data, and missing `primaryName` values (including IMDb's `\N` marker, not just true nulls).

```sql
SELECT COUNT(*) AS total_rows FROM name_basics;

SELECT * FROM name_basics LIMIT 5;

SELECT COUNT(*) AS missing_primaryName
FROM name_basics
WHERE primaryName IS NULL OR primaryName = '\N';
```
Result (sample):

| nconst | primaryName | birthYear | deathYear | primaryProfession | knownForTitles |
|---|---|---|---|---|---|
| nm0000001 | Fred Astaire | 1899 | 1987 | actor, miscellaneous, producer | tt0072308, tt0050419, ... |
| nm0000002 | Lauren Bacall | 1924 | 2014 | actress, miscellaneous, soundtrack | tt0037382, tt0075213, ... |
| nm0000003 | Brigitte Bardot | 1934 | 2025 | actress, music_department, producer | tt0057345, tt0049189, ... |
| nm0000004 | John Belushi | 1949 | 1982 | actor, writer, music_department | tt0072562, tt0077975, ... |
| nm0000005 | Ingmar Bergman | 1918 | 2007 | writer, director, actor | tt0050986, tt0083922, ... |

2. **title_ratings**: checked row count and the range of `averageRating` and `numVotes`, to spot anything outside expected bounds.

```sql
SELECT COUNT(*) AS total_rows FROM title_ratings;

SELECT
    MIN(averageRating) AS min_rating,
    MAX(averageRating) AS max_rating,
    MIN(numVotes) AS min_votes,
    MAX(numVotes) AS max_votes
FROM title_ratings;
```
Result:

| min_rating | max_rating | min_votes | max_votes |
|---|---|---|---|
| 1.0 | 10.0 | 5 | 3,228,898 |

3. **title_basics**: checked row count, the breakdown of `titleType` values, how many rows have a missing `startYear`, and the year range for movies only.

```sql
SELECT COUNT(*) AS total_rows FROM title_basics;

SELECT titleType, COUNT(*) AS row_count
FROM title_basics
GROUP BY titleType
ORDER BY row_count DESC;

SELECT COUNT(*) AS missing_startYear
FROM title_basics
WHERE startYear = '\N';

SELECT MIN(startYear) AS earliest_year, MAX(startYear) AS latest_year
FROM title_basics
WHERE titleType = 'movie' AND startYear != '\N';
```
Result: ...

4. **title_principals** (large table counts and samples only): checked row count and which roles/categories exist (actor, director, writer, etc.).

```sql
SELECT COUNT(*) AS total_rows FROM title_principals;

SELECT category, COUNT(*) AS row_count
FROM title_principals
GROUP BY category
ORDER BY row_count DESC;
```
Result: ...

5. **tmdb**: checked row count, column names (since TMDB's schema isn't documented like IMDb's), missing/zero budget and revenue counts, and the release year range.

```sql
SELECT COUNT(*) AS total_rows FROM tmdb;

PRAGMA table_info(tmdb);

SELECT
    SUM(CASE WHEN budget IS NULL OR budget = 0 THEN 1 ELSE 0 END) AS missing_or_zero_budget,
    SUM(CASE WHEN revenue IS NULL OR revenue = 0 THEN 1 ELSE 0 END) AS missing_or_zero_revenue,
    SUM(CASE WHEN release_date IS NULL THEN 1 ELSE 0 END) AS missing_release_date
FROM tmdb;

SELECT
    MIN(strftime('%Y', release_date)) AS earliest_year,
    MAX(strftime('%Y', release_date)) AS latest_year
FROM tmdb;
```
Result: ...

6. **Filtering**: based on the exploration above, `title_basics` and `tmdb` were filtered to movies within a 2013 2024 margin (a wider window than the current 2014 2023 problem statement range, in case the final range changes later), and the filtered results were exported as intermediate cleaned files, `title_basics_filtered.csv` and `tmdb_filtered.csv`, and stored in `datasets/cleaned/`.

```sql
SELECT * 
FROM title_basics 
WHERE titleType = 'movie' AND startYear BETWEEN 2013 AND 2024;

SELECT *
FROM tmdb
WHERE strftime('%Y', release_date) BETWEEN '2013' AND '2024';
```
Result: confirmed row counts shown when the CSV files are loaded into pandas below.

The exported CSV files were then loaded into pandas below to continue the cleaning process.


In [3]:
import pandas as pd
from pathlib import Path

CLEANED_PATH = Path("datasets/cleaned")

# Loaded from the CSV files exported by DB Browser for SQLite (see Step 2 above)
title_basics_filtered = pd.read_csv(CLEANED_PATH / "title_basics_filtered.csv")
tmdb_filtered = pd.read_csv(CLEANED_PATH / "tmdb_filtered.csv")

print(f"title_basics_filtered: {title_basics_filtered.shape[0]:,} rows, {title_basics_filtered.shape[1]} columns")
print(f"tmdb_filtered: {tmdb_filtered.shape[0]:,} rows, {tmdb_filtered.shape[1]} columns")


title_basics_filtered: 227,265 rows, 9 columns
tmdb_filtered: 486,063 rows, 24 columns


### Step 4: Check Data Quality

Both filtered tables were checked for missing values and duplicate rows before merging.


In [4]:
# .isnull().sum() counts missing values per column, .duplicated().sum()
# counts fully duplicate rows - checked here, before merging, so any issue
# found is easy to trace back to its source table
for name, df in [("title_basics_filtered", title_basics_filtered), ("tmdb_filtered", tmdb_filtered)]:
    print(f"--- {name} ---")
    print(f"Shape: {df.shape[0]:,} rows, {df.shape[1]} columns")
    print(f"Duplicate rows: {df.duplicated().sum():,}")
    print(f"Missing values:\n{df.isnull().sum()}\n")

--- title_basics_filtered ---
Shape: 227,265 rows, 9 columns
Duplicate rows: 0
Missing values:
tconst            0
titleType         0
primaryTitle      1
originalTitle     1
isAdult           0
startYear         0
endYear           0
runtimeMinutes    0
genres            0
dtype: int64

--- tmdb_filtered ---
Shape: 486,063 rows, 24 columns
Duplicate rows: 96
Missing values:
id                           0
title                        3
vote_average                 0
vote_count                   0
status                       0
release_date                 0
revenue                      0
runtime                      0
adult                        0
backdrop_path           309754
budget                       0
homepage                407882
imdb_id                 279802
original_language            0
original_title               3
overview                 85655
popularity                   0
poster_path              92021
tagline                 412907
genres                  170473
pr

### Step 5: Review IMDb Title Columns

IMDb stores missing values as the literal text `\N`, not a true empty value, so it's replaced with a real missing value first. Any column that is missing in more than 90% of rows is dropped, since it can't meaningfully support analysis at that point.


In [5]:
# IMDb stores missing values as the literal text '\N', not a true NaN,
# so it's replaced with a real missing value before counting anything
clean = title_basics_filtered.replace("\\N", pd.NA)
missing_pct = (clean.isnull().sum() / len(clean) * 100).round(2)
print(missing_pct)

# A column is dropped only if it's mostly empty (over 90% missing) -
# at that point it can't meaningfully support any analysis
cols_to_drop = missing_pct[missing_pct > 90].index.tolist()
print(f"\nDropped (mostly missing): {cols_to_drop}")

title_basics_clean = clean.drop(columns=cols_to_drop)

# IMDb's original column names use camelCase; renamed to snake_case here
# for consistency with TMDB's naming before the two tables are merged
title_basics_clean = title_basics_clean.rename(columns={
    "primaryTitle": "primary_title",
    "originalTitle": "original_title",
    "isAdult": "is_adult",
    "startYear": "start_year",
    "runtimeMinutes": "runtime_minutes"
})

print(f"Remaining columns: {title_basics_clean.columns.tolist()}")

tconst              0.00
titleType           0.00
primaryTitle        0.00
originalTitle       0.00
isAdult             0.00
startYear           0.00
endYear           100.00
runtimeMinutes     23.66
genres              4.06
dtype: float64

Dropped (mostly missing): ['endYear']
Remaining columns: ['tconst', 'titleType', 'primary_title', 'original_title', 'is_adult', 'start_year', 'runtime_minutes', 'genres']


### Step 6: Review TMDB Columns

The same missing value check was applied here. Columns unrelated to this project's objectives (genre, budget, previous lead experience, and Reported Return vs. IMDb rating) were also dropped: `homepage`, `backdrop_path`, `poster_path`, `tagline`, `overview`, `keywords`.

`title`, `original_title`, `genres`, and `runtime` exist in both tables, so the TMDB versions are prefixed with `tmdb_` to avoid confusion after merging.


In [6]:
missing_pct = (tmdb_filtered.isnull().sum() / len(tmdb_filtered) * 100).round(2)

# not_relevant: complete columns that just don't relate to this project's
# objectives (image paths, free text descriptions have no analytical value)
not_relevant = ["homepage", "backdrop_path", "poster_path", "tagline", "overview", "keywords"]
cols_to_drop = list(set(not_relevant + missing_pct[missing_pct > 90].index.tolist()))
print(f"Dropped: {cols_to_drop}")

tmdb_clean = tmdb_filtered.drop(columns=[c for c in cols_to_drop if c in tmdb_filtered.columns])

# title, original_title, genres, and runtime exist in BOTH tables - left
# unrenamed, merging would silently overwrite one version with the other
tmdb_clean = tmdb_clean.rename(columns={
    "title": "tmdb_title",
    "original_title": "tmdb_original_title",
    "genres": "tmdb_genres",
    "runtime": "tmdb_runtime"
})

print(f"Remaining columns: {tmdb_clean.columns.tolist()}")


Dropped: ['backdrop_path', 'overview', 'keywords', 'poster_path', 'homepage', 'tagline']
Remaining columns: ['id', 'tmdb_title', 'vote_average', 'vote_count', 'status', 'release_date', 'revenue', 'tmdb_runtime', 'adult', 'budget', 'imdb_id', 'original_language', 'tmdb_original_title', 'popularity', 'tmdb_genres', 'production_companies', 'production_countries', 'spoken_languages']


### Step 7: Review the Remaining IMDb Tables

`title_ratings` and `name_basics` are small enough to load fully. `title_principals` is too large (4.4 GB, 101M+ rows), so only its row count and a small sample are checked here; it will be queried directly when needed later.


In [7]:
import sqlite3
from pathlib import Path

raw_db = Path("datasets/raw") / "raw_data.db"

if raw_db.exists():
    conn = sqlite3.connect(raw_db)

    title_ratings_full = pd.read_sql("SELECT * FROM title_ratings", conn)
    name_basics_full = pd.read_sql("SELECT * FROM name_basics", conn)
    principals_count = pd.read_sql(
        "SELECT COUNT(*) AS n FROM title_principals", conn
    )["n"][0]

    print(f"title_ratings: {title_ratings_full.shape[0]:,} rows, {title_ratings_full.duplicated().sum():,} duplicates")
    print(f"name_basics: {name_basics_full.shape[0]:,} rows, {name_basics_full.duplicated().sum():,} duplicates")
    print(f"title_principals: {principals_count:,} rows (not fully loaded)")

    conn.close()
else:
    print("Exploratory SQLite check skipped. The submitted raw source files and cleaned intermediate outputs document this step; raw_data.db is an optional working database, not a required submission dependency.")


title_ratings: 1,709,630 rows, 0 duplicates
name_basics: 15,601,022 rows, 0 duplicates
title_principals: 101,425,963 rows (not fully loaded)


### Step 8: Merge IMDb and TMDB

IMDb `tconst` was matched to TMDB `imdb_id`. An inner join was used for the film level financial sample because the reported-return measure requires usable reported budget and revenue.

The documented cleaning funnel is:

**227,265 IMDb title records after the initial filter + 486,063 TMDB rows → 102,868 matched rows → 2,826 rows after usable reported budget/revenue and below-$1,000 exclusions → 2,465 unique IMDb films released from 2014 to 2023.**

IMDb ratings were added for descriptive context but were not required as a before funding model predictor.


In [8]:
# how="inner" keeps only films present in BOTH tables - a film missing
# from one side can't produce a budget/revenue/rating pair anyway
merged = title_basics_clean.merge(tmdb_clean, left_on="tconst", right_on="imdb_id", how="inner")

# Ratings joined separately with how="left": a missing rating isn't a
# reason to drop a film the way a missing budget/revenue is
merged = merged.merge(title_ratings_full, on="tconst", how="left")

print(f"Merged: {merged.shape[0]:,} rows, {merged.shape[1]} columns")
print(f"Missing averageRating: {merged['averageRating'].isnull().sum():,}")

Merged: 102,868 rows, 28 columns
Missing averageRating: 12,943


### Step 9: Handle Invalid Budget and Revenue Values

Rows with missing or zero budget/revenue were excluded because the reported-return calculation requires both values. Very small reported financial values also produced implausibly extreme return ratios, so the project applied a **$1,000 minimum to both reported budget and reported revenue**.

The $1,000 rule is a data quality rule for this project, not an industry definition of a valid film budget or revenue value. The excluded alternatives were not reclassified as genuine zero-revenue films.


In [9]:
# A budget or revenue of exactly 0 in TMDB means the figure was never
# reported, not that it was genuinely zero - so these rows are excluded
invalid = ((merged["budget"] <= 0) | (merged["revenue"] <= 0)).sum()
print(f"Rows with missing/zero budget or revenue: {invalid:,}")

merged_valid = merged[(merged["budget"] > 0) & (merged["revenue"] > 0)].copy()

# Some remaining values were still unrealistically low (as low as $1-$500),
# producing extreme ROI outliers - a $1,000 minimum was agreed with the
# instructional team as a cutoff for these likely data entry errors
below_threshold = ((merged_valid["budget"] < 1000) | (merged_valid["revenue"] < 1000)).sum()
print(f"Rows below the $1,000 threshold: {below_threshold:,}")

merged_valid = merged_valid[(merged_valid["budget"] >= 1000) & (merged_valid["revenue"] >= 1000)].copy()
print(f"Remaining rows: {merged_valid.shape[0]:,}")

Rows with missing/zero budget or revenue: 99,931
Rows below the $1,000 threshold: 111
Remaining rows: 2,826


### Step 10: Remove Duplicates and Calculate Reported Return

Duplicate films were checked using the IMDb title identifier (`tconst` / final `imdb_id`). After the cleaning and date filters, the final analytical dataset contains **2,465 unique films and zero duplicate final IMDb IDs**.

The technical dataset column remains named `ROI` for compatibility with the existing code, but the presentation and interpretation use **Reported Return**:

`Reported Return = reported revenue minus production budget, divided by production budget`

A value below 0 means reported revenue is reported revenue below production budget. This is **not actual investor profit or true break-even**, because major costs and revenue sharing items are not included.


In [10]:
# subset=["tconst"] catches the same film appearing twice even if some
# other column differs slightly between the two rows
dupes = merged_valid.duplicated(subset=["tconst"]).sum()
print(f"Duplicate rows: {dupes:,}")

merged_valid = merged_valid.drop_duplicates(subset=["tconst"], keep="first")

# Technical column name ROI stores the reported return measure:
# (reported revenue - production budget) / production budget.
# It is not actual net profit because other film costs are excluded.
# The calculation is safe here because every row has a budget of at least $1,000
merged_valid["ROI"] = (merged_valid["revenue"] - merged_valid["budget"]) / merged_valid["budget"]

print(f"Final row count: {merged_valid.shape[0]:,}")
print(merged_valid["ROI"].describe())

Duplicate rows: 0
Final row count: 2,826
count    2826.000000
mean        3.451244
std        19.935203
min        -0.999929
25%        -0.408986
50%         0.713553
75%         2.757454
max       525.469870
Name: ROI, dtype: float64


### Step 11: Finalize the Analysis Columns

Merging left several duplicated columns (two title columns, two runtime columns, two rating systems, etc.), since the same information exists in both sources under different names. The dataset is narrowed to one version of each field, columns unrelated to the project's objectives are dropped, and the rest are renamed for clarity.

`is_adult` was also dropped: every remaining row has the same value (0), so the column carries no information after filtering.

`production_companies` and `popularity` are kept even though they're outside the current four objectives, since they may support an additional objective later.

Dropped as duplicates: `original_title`, `tmdb_title`, `tmdb_original_title`, `id`, `imdb_id` (TMDB's copy `tconst` is kept instead), `adult`, `tmdb_runtime`, `tmdb_genres`, `vote_average`, `vote_count`, `is_adult`.

Dropped as not relevant to the objectives: `endYear` (empty for all films), `status`, `release_date`, `production_countries`, `spoken_languages`.


In [11]:
# Maps each column to keep -> its final, clearer name. Columns not
# listed here (duplicates, or unrelated to the project's objectives) are
# dropped simply by not being included in this selection
final_columns = {
    "tconst": "imdb_id",
    "primary_title": "title",
    "start_year": "release_year",
    "runtime_minutes": "runtime_minutes",
    "genres": "genres",
    "budget": "budget",
    "revenue": "revenue",
    "ROI": "ROI",
    "averageRating": "imdb_rating",
    "numVotes": "imdb_vote_count",
    "original_language": "original_language",
    "production_companies": "production_companies",
    "popularity": "popularity",
}

merged_valid = merged_valid[list(final_columns.keys())].rename(columns=final_columns)
print(f"Final columns ({len(merged_valid.columns)}): {merged_valid.columns.tolist()}")

Final columns (13): ['imdb_id', 'title', 'release_year', 'runtime_minutes', 'genres', 'budget', 'revenue', 'ROI', 'imdb_rating', 'imdb_vote_count', 'original_language', 'production_companies', 'popularity']


### Step 12: Check Data Types

Before exporting, each column's data type was checked to confirm numeric fields (`budget`, `revenue`, `release_year`, `runtime_minutes`, `imdb_rating`, `imdb_vote_count`, `Reported Return`) are actually stored as numbers, not text, and that `budget`/`revenue` don't contain mixed formats.

`budget` and `revenue` are assumed to be in USD, based on TMDB's documented convention for these fields. This can't be independently verified for every row, so it's noted as an assumption rather than a confirmed fact (it's checked with a quick sanity comparison against two well known films with public, real world budget figures).

`runtime_minutes` was originally stored as text due to IMDb's `\N` missing value marker mixed in with numeric values. It was converted to a numeric type using `pd.to_numeric(..., errors="coerce")`, which left 28 films with a missing runtime after conversion.


**Runtime note:** An earlier intermediate dataset contained 28 missing runtime values. After the final 2014 to 2023 sample was created, **23 of the 2,465 final films** had missing runtime. The final presentation and model documentation use the final sample count of 23.


In [12]:
# runtime_minutes came in as text because some values contain '\N' (IMDb's
# missing marker). pd.to_numeric with errors="coerce" converts valid
# numbers properly and turns any remaining '\N' into a real NaN
merged_valid["runtime_minutes"] = pd.to_numeric(merged_valid["runtime_minutes"], errors="coerce")

print(merged_valid.dtypes)

# budget/revenue should be int64 or float64 - if either shows up as
# "object" (text), it means some rows contain non numeric characters
non_numeric = merged_valid[["budget", "revenue"]].select_dtypes(include="object").columns.tolist()
print(f"\nNon-numeric budget/revenue columns (should be empty): {non_numeric}")

# Sanity check: compare two well known films against their real world
# budgets, to see whether the $ values in this dataset look plausible
known_films = merged_valid[merged_valid["title"].isin(["Barbie", "Oppenheimer"])]
print(f"\nKnown-budget sanity check:\n{known_films[['title', 'release_year', 'budget', 'revenue']]}")

imdb_id                     str
title                       str
release_year              int64
runtime_minutes         float64
genres                      str
budget                    int64
revenue                   int64
ROI                     float64
imdb_rating             float64
imdb_vote_count         float64
original_language           str
production_companies        str
popularity              float64
dtype: object

Non-numeric budget/revenue columns (should be empty): []

Known-budget sanity check:
             title  release_year     budget     revenue
93016       Barbie          2023  145000000  1428545028
93126  Oppenheimer          2023  100000000   933700000


### Step 13: Export the Cleaned Dataset


In [14]:
from pathlib import Path

CLEANED_PATH = Path("datasets/cleaned")
CLEANED_PATH.mkdir(parents=True, exist_ok=True)

merged_valid.to_csv(
    CLEANED_PATH / "movies_cleaned_filtered.csv",
    index=False
)

print(
    f"Saved: {merged_valid.shape[0]:,} rows, "
    f"{merged_valid.shape[1]} columns"
)

Saved: 2,826 rows, 13 columns


### Step 14: Link Cast and Crew Data

`title_principals` is too large (4.4 GB, 101M+ rows) to join efficiently in pandas. Instead, the join was done directly in DB Browser for SQLite:

1. `movies_cleaned_filtered.csv` (the final 2,826 films) was imported into `raw_data.db` as a new table, `final_films`.
2. The following query joined `title_principals` to `final_films` on the film ID, filtered to actors, actresses, directors, and writers, and pulled in each person's name from `name_basics`:

```sql
SELECT tp.tconst AS imdb_id, tp.category, tp.ordering,
       nb.nconst, nb.primaryName, nb.knownForTitles
FROM title_principals tp
JOIN final_films ff ON tp.tconst = ff.imdb_id
JOIN name_basics nb ON tp.nconst = nb.nconst
WHERE tp.category IN ('actor', 'actress', 'director', 'writer')
```

3. The result was exported as `cast_crew_filtered.csv`.

This step only confirms that cast and crew data can be linked to the final film list it does not yet calculate a Previous Lead Experience value, since that depends on the proxy definition noted under Unresolved data quality issues.


In [15]:
# Loaded from the CSV exported by DB Browser for SQLite (look at Step 12 above)
cast_crew = pd.read_csv(CLEANED_PATH / "cast_crew_filtered.csv")

print(f"Shape: {cast_crew.shape[0]:,} rows, {cast_crew.shape[1]} columns")
print(f"Unique films linked: {cast_crew['imdb_id'].nunique():,} out of {merged_valid.shape[0]:,}")
print(f"\nBy category:\n{cast_crew['category'].value_counts()}")

Shape: 37,483 rows, 6 columns
Unique films linked: 2,825 out of 2,826

By category:
category
actor       18281
actress      9789
writer       6353
director     3060
Name: count, dtype: int64


In [16]:
import sqlite3
from pathlib import Path

raw_db = Path("datasets/raw") / "raw_data.db"

if raw_db.exists():
    conn = sqlite3.connect(raw_db)

    barbie_rows = merged_valid.loc[merged_valid["title"] == "Barbie", "imdb_id"]
    if len(barbie_rows) > 0:
        barbie_id = barbie_rows.iloc[0]
        query = f"""
        SELECT tp.ordering, tp.category, nb.primaryName
        FROM title_principals tp
        JOIN name_basics nb ON tp.nconst = nb.nconst
        WHERE tp.tconst = '{barbie_id}'
        ORDER BY tp.ordering
        """
        print(pd.read_sql(query, conn))
    else:
        print("Barbie sanity check skipped because the title was not found in the current cleaned sample.")

    conn.close()
else:
    print("Optional cast and crew sanity check skipped because raw_data.db is not required for the submitted analysis workflow.")


    ordering             category      primaryName
0          1              actress    Margot Robbie
1          2                actor     Ryan Gosling
2          3              actress         Issa Rae
3          4              actress    Kate McKinnon
4          5              actress  Alexandra Shipp
5          6              actress      Emma Mackey
6          7              actress         Hari Nef
7          8              actress    Sharon Rooney
8          9              actress   Ana Cruz Kayne
9         10              actress        Ritu Arya
10        11             director     Greta Gerwig
11        12               writer     Greta Gerwig
12        13               writer    Noah Baumbach
13        14             producer     Tom Ackerley
14        15             producer   Robbie Brenner
15        16             producer     David Heyman
16        17             producer    Margot Robbie
17        18             composer      Mark Ronson
18        19             compos

## Descriptive Analysis

The following analysis follows the final presentation structure. The project uses **Reported Return** as the financial measure and **Previous Lead Experience** as the audience facing name for the technical `star_power` field.

### Objective 1: Before Funding Factors and Reported Return
The analysis compares the individual linear relationships of production budget, previous lead experience, and historical genre return with reported return. All three correlations are close to zero, so they show weak individual linear relationships in this selected sample.

### Objective 2: Below Budget Rate by Genre
Genre comparisons use the percentage of films whose reported revenue is reported revenue below production budget. Only genres with at least 25 films are reported in the main presentation. Films can belong to more than one genre.

### Objective 3: Below Budget Rate by Budget Level
Films are grouped into the budget bands used in the presentation. The comparison describes how the below budget rate varies across those groups. It does not establish that increasing a film budget reduces financial risk.

### Objective 4: Previous Lead Experience
Films are compared according to whether the lead had previous lead experience. The presentation uses median reported return and below budget rate. The groups show different median reported return but very similar below budget rates, so previous lead experience is not treated as a clear standalone signal.

### Objective 5: Director and Writer Prior History
Director and writer history uses earlier year films available inside the selected financial dataset. Same year releases are excluded. These fields do not represent complete careers and are available for only part of the final sample.

### Combined Model
The regression considers the factors together and adds production company controls. The outcome is limited at the 1st and 99th percentile cutoffs while all films are kept. The model is exploratory and was not tested on new films.


In [17]:
import pandas as pd
from pathlib import Path

CLEANED_PATH = Path("datasets/cleaned")
merged_valid = pd.read_csv(CLEANED_PATH / "movies_cleaned_filtered.csv")

# Filter to the problem statement's exact year range
df_ps = merged_valid[(merged_valid["release_year"] >= 2014) & (merged_valid["release_year"] <= 2023)].copy()
print(f"Films in 2014 to 2023: {df_ps.shape[0]:,}")

genres_split = df_ps["genres"].str.get_dummies(sep=",")
print(f"Number of distinct genres: {genres_split.shape[1]}")

genre_corr = {genre: genres_split[genre].corr(df_ps["ROI"]) for genre in genres_split.columns}
genre_corr = pd.Series(genre_corr).sort_values(key=abs, ascending=False)

print("\nCorrelation of each genre with ROI (sorted by strength):")
print(genre_corr)

print(f"\nMax absolute correlation: {genre_corr.abs().max():.4f}")
print(f"Genre with strongest correlation: {genre_corr.abs().idxmax()}")

Films in 2014 to 2023: 2,465
Number of distinct genres: 22

Correlation of each genre with ROI (sorted by strength):
Horror         0.091567
Music          0.049418
Mystery        0.040916
Crime         -0.032649
Action        -0.027095
Biography     -0.026468
History       -0.024326
Fantasy       -0.023644
Thriller       0.021133
Drama         -0.019754
Adventure     -0.019473
Romance       -0.018253
War           -0.016871
Sci-Fi        -0.015618
Musical       -0.013689
Sport         -0.013502
Family        -0.012257
Western       -0.011448
Documentary    0.008082
Animation     -0.003201
News          -0.002594
Comedy        -0.000191
dtype: float64

Max absolute correlation: 0.0916
Genre with strongest correlation: Horror


### Supporting Calculation
The following code cell supports the descriptive analysis above and retains the original technical field names required by the dataset.


In [18]:
# df_ps was already created in Step 13; reused here
budget_corr = df_ps["budget"].corr(df_ps["ROI"])
print(f"Correlation budget vs ROI: {budget_corr:.4f}")

# For context: descriptive stats on budget within this filtered subset
print(f"\nBudget range: ${df_ps['budget'].min():,} to ${df_ps['budget'].max():,}")
print(f"Median budget: ${df_ps['budget'].median():,.0f}")

Correlation budget vs ROI: -0.0439

Budget range: $1,000 to $460,000,000
Median budget: $10,000,000


### Supporting Calculation
The following code cell supports the descriptive analysis above and retains the original technical field names required by the dataset.


### Supporting Calculation
The following code cell supports the descriptive analysis above and retains the original technical field names required by the dataset.


In [19]:
# Standalone cell: loads and imports everything it needs, doesn't depend on
# any earlier cell having been run first
import pandas as pd
from pathlib import Path

CLEANED_PATH = Path("datasets/cleaned")

# Load the cleaned film data fresh, and filter to the problem statement's window
movies = pd.read_csv(CLEANED_PATH / "movies_cleaned_filtered.csv")
df_ps = movies[(movies["release_year"] >= 2014) & (movies["release_year"] <= 2023)].copy()

# Load the three "prior films" files. Each one has ONE ROW PER PERSON
# credited in that role on a film, not one row per film, so a film with two
# directors has two rows here. Merging this directly onto df_ps (one row per
# film) would duplicate that film's row once per person, and the flag value
# could even differ between the duplicate rows.
director_prior = pd.read_csv(CLEANED_PATH / "director_prior_counts.csv")
writer_prior = pd.read_csv(CLEANED_PATH / "writer_prior_counts.csv")
lead_prior = pd.read_csv(CLEANED_PATH / "lead_prior_counts.csv")

# Collapse each file to one row per imdb_id BEFORE merging. max() on a 0/binary
# count works like "any": if any credited person had >= 1 prior film, the
# film level value is at least 1, matching the original Step 15 "any" logic
director_prior_agg = director_prior.groupby("imdb_id")["prior_director_films"].max().reset_index()
writer_prior_agg = writer_prior.groupby("imdb_id")["prior_writer_films"].max().reset_index()
lead_prior_agg = lead_prior.groupby("imdb_id")["prior_lead_films"].max().reset_index()

rows_before = len(df_ps)

# Merge each one onto df_ps by imdb_id. Since each *_agg table now has at
# most one row per imdb_id, this is a clean one to one merge with no fan out
df_ps = df_ps.merge(director_prior_agg, on="imdb_id", how="left")
df_ps = df_ps.merge(writer_prior_agg, on="imdb_id", how="left")
df_ps = df_ps.merge(lead_prior_agg, on="imdb_id", how="left")

# Row count sanity check: the merge must not have changed the number of
# films. If it did, something upstream still has duplicate imdb_id values
assert len(df_ps) == rows_before, f"Row count changed after merge: {rows_before} -> {len(df_ps)}"
print(f"Row count unchanged after merge: {len(df_ps):,} films")

# Recurring means having at least 1 prior film in that role before this
# film's release. fillna(0) handles films with no director, writer, or lead
# data at all, since no prior data correctly means "not recurring", not unknown
df_ps["rec_director"] = (df_ps["prior_director_films"].fillna(0) >= 1).astype(int)
df_ps["rec_writer"] = (df_ps["prior_writer_films"].fillna(0) >= 1).astype(int)
df_ps["star_power"] = (df_ps["prior_lead_films"].fillna(0) >= 1).astype(int)

# Drop the raw count columns now that the 0/1 flags are built. This keeps
# the final table matching the same shape as before
df_ps = df_ps.drop(columns=["prior_director_films", "prior_writer_films", "prior_lead_films"])

print(df_ps[["star_power", "rec_director", "rec_writer"]].apply(pd.Series.value_counts))

# Extra check specific to this dataset: confirm imdb_id is still unique
assert df_ps["imdb_id"].is_unique, "Duplicate imdb_id values found after merge"

# Split the comma separated genres column into one 0/1 column per genre
genres_split = df_ps["genres"].str.get_dummies(sep=",")

# Prefix with "genre_" so these columns are clearly distinguishable from the
# film's original descriptive columns (title, budget, etc.)
genres_split = genres_split.add_prefix("genre_")

# Attach the genre dummy columns to the main table
df_ps = pd.concat([df_ps, genres_split], axis=1)

print(f"Added {genres_split.shape[1]} genre columns: {genres_split.columns.tolist()}")

# Export the fully merged table under a new name, keeping the original
# movies_cleaned_filtered.csv untouched as a fallback
df_ps.to_csv(CLEANED_PATH / "movies_cleaned_final.csv", index=False)
print(f"Saved: {df_ps.shape[0]:,} rows, {df_ps.shape[1]} columns")


Row count unchanged after merge: 2,465 films
   star_power  rec_director  rec_writer
1        2303          2240        2259
0         162           225         206
Added 22 genre columns: ['genre_Action', 'genre_Adventure', 'genre_Animation', 'genre_Biography', 'genre_Comedy', 'genre_Crime', 'genre_Documentary', 'genre_Drama', 'genre_Family', 'genre_Fantasy', 'genre_History', 'genre_Horror', 'genre_Music', 'genre_Musical', 'genre_Mystery', 'genre_News', 'genre_Romance', 'genre_Sci-Fi', 'genre_Sport', 'genre_Thriller', 'genre_War', 'genre_Western']
Saved: 2,465 rows, 38 columns


In [20]:
# Standalone cell: loads and imports everything it needs, doesn't depend on
# any earlier cell having been run first. Reads the corrected file exported
# by Step 16 above, so run that cell first at least once so the file exists
# on disk.
import pandas as pd
from pathlib import Path

CLEANED_PATH = Path("datasets/cleaned")
df_ps = pd.read_csv(CLEANED_PATH / "movies_cleaned_final.csv")

# Recalculate genre correlations on the corrected df_ps
genre_cols = [c for c in df_ps.columns if c.startswith("genre_")]
genre_corr_revised = df_ps[genre_cols].corrwith(df_ps["ROI"])
genre_max_corr_revised = genre_corr_revised.abs().max()

budget_corr_revised = df_ps["budget"].corr(df_ps["ROI"])
star_corr_revised = df_ps["star_power"].corr(df_ps["ROI"])

factor_correlations_revised = {
    "genre (Horror, strongest)": genre_max_corr_revised,
    "budget": abs(budget_corr_revised),
    "star_power": abs(star_corr_revised),
}
avg_correlation_revised = sum(factor_correlations_revised.values()) / len(factor_correlations_revised)

print("Corrected correlation by factor (three factor problem statement):")
for factor, value in factor_correlations_revised.items():
    print(f"  {factor}: {value:.4f}")
print(f"\nCorrected average across the three factors: {avg_correlation_revised:.4f}")

# For reference only: rec_director and rec_writer, not part of the original
# three factor problem statement, but computed the same corrected way
rec_director_corr = df_ps["rec_director"].corr(df_ps["ROI"])
rec_writer_corr = df_ps["rec_writer"].corr(df_ps["ROI"])
print(f"\nFor reference:")
print(f"  rec_director: {abs(rec_director_corr):.4f}")
print(f"  rec_writer: {abs(rec_writer_corr):.4f}")


Corrected correlation by factor (three factor problem statement):
  genre (Horror, strongest): 0.0916
  budget: 0.0439
  star_power: 0.0333

Corrected average across the three factors: 0.0563

For reference:
  rec_director: 0.0586
  rec_writer: 0.0786


## Prior Performance Features

To support the analysis of factors available before a film funding decision, three historical performance features were created:

- **Historical Genre Reported Return:** The average Reported Return of films released in prior years within the same genre(s).
- **Historical Director Reported Return:** The average Reported Return of films released in prior years by the same director(s).
- **Historical Writer Reported Return:** The average Reported Return of films released in prior years by the same writer(s).

Only films released **before the current film's release year** were included in each calculation. Films from the same year and future years were excluded to prevent look ahead bias and ensure that each feature represents information that could have been available before the film's release.

For films associated with multiple genres, directors, or writers, the historical Reported Return was calculated separately for each associated category or person and then averaged at the film level.

Where no prior observations were available, the historical value was retained as missing rather than imputed using future information.


In [21]:
import pandas as pd
import numpy as np

# Load the cleaned datasets
movies = pd.read_csv("datasets/cleaned/movies_cleaned_final.csv")
cast_crew = pd.read_csv("datasets/cleaned/cast_crew_filtered.csv")

# Check the size of each dataset
print("Movies shape:", movies.shape)
print("Cast/Crew shape:", cast_crew.shape)

Movies shape: (2465, 38)
Cast/Crew shape: (37483, 6)


### Prior Genre Reported Return

This feature measures the average Reported Return of films released in previous years within the same genre.

For films with multiple genres, historical Reported Return is calculated for each genre separately and then averaged at the film level. Films released in the same year or later are excluded.


In [22]:
# Keep only the columns needed for genre analysis
genre_data = movies[
    ["imdb_id", "release_year", "ROI", "genres"]
].copy()

# Split films with multiple genres into separate rows
genre_data["genre"] = genre_data["genres"].fillna("").str.split(",")
genre_data = genre_data.explode("genre")

# Clean genre names
genre_data["genre"] = genre_data["genre"].str.strip()
genre_data = genre_data[genre_data["genre"] != ""]

# Calculate total ROI and number of films for each genre and year
genre_year = (
    genre_data
    .groupby(["genre", "release_year"])
    .agg(
        sum_roi=("ROI", "sum"),
        film_count=("ROI", "count")
    )
    .reset_index()
)

# Sort by genre and year before calculating historical values
genre_year = genre_year.sort_values(
    ["genre", "release_year"]
)

# Calculate cumulative ROI from previous years only
genre_year["prior_roi_sum"] = (
    genre_year.groupby("genre")["sum_roi"].cumsum()
    - genre_year["sum_roi"]
)

# Calculate the number of previous films in each genre
genre_year["prior_film_count"] = (
    genre_year.groupby("genre")["film_count"].cumsum()
    - genre_year["film_count"]
)

# Calculate historical ROI for each genre
genre_year["historical_genre_roi_component"] = (
    genre_year["prior_roi_sum"]
    / genre_year["prior_film_count"]
)

# Add the historical genre ROI back to each film
genre_data = genre_data.merge(
    genre_year[
        [
            "genre",
            "release_year",
            "historical_genre_roi_component"
        ]
    ],
    on=["genre", "release_year"],
    how="left"
)

# Average historical ROI across all genres assigned to each film
historical_genre = (
    genre_data
    .groupby("imdb_id")["historical_genre_roi_component"]
    .mean()
    .rename("historical_genre_roi")
)

### Prior Director and Writer Reported Return

The cast and crew dataset was linked to the movie dataset using `imdb_id`.

Directors and writers were identified using their IMDb `nconst` identifiers. Historical Reported Return was calculated using only films released before the current film's release year.

If a film had multiple directors or writers, the available historical Reported Return values were averaged at the film level.


In [23]:
# Keep director and writer records only
crew_roles = cast_crew[
    cast_crew["category"].isin(["director", "writer"])
][
    ["imdb_id", "category", "nconst", "primaryName"]
].drop_duplicates()

# Add each film's release year and ROI
crew_movies = crew_roles.merge(
    movies[["imdb_id", "release_year", "ROI"]],
    on="imdb_id",
    how="inner"
)

# Aggregate each person's films by year
person_year = (
    crew_movies
    .groupby(["category", "nconst", "release_year"])
    .agg(
        sum_roi=("ROI", "sum"),
        film_count=("ROI", "count")
    )
    .reset_index()
)

# Sort records chronologically for each person
person_year = person_year.sort_values(
    ["category", "nconst", "release_year"]
)

# Calculate cumulative ROI from previous years only
person_year["prior_roi_sum"] = (
    person_year
    .groupby(["category", "nconst"])["sum_roi"]
    .cumsum()
    - person_year["sum_roi"]
)

# Count previous films for each person
person_year["prior_film_count"] = (
    person_year
    .groupby(["category", "nconst"])["film_count"]
    .cumsum()
    - person_year["film_count"]
)

# Calculate each person's historical ROI
person_year["historical_person_roi"] = (
    person_year["prior_roi_sum"]
    / person_year["prior_film_count"]
)

# Add historical performance back to each film person record
crew_movies = crew_movies.merge(
    person_year[
        [
            "category",
            "nconst",
            "release_year",
            "historical_person_roi"
        ]
    ],
    on=["category", "nconst", "release_year"],
    how="left"
)

# Average multiple directors or writers at the film level
historical_people = (
    crew_movies
    .groupby(["imdb_id", "category"])["historical_person_roi"]
    .mean()
    .unstack()
)

# Rename the final columns
historical_people = historical_people.rename(
    columns={
        "director": "historical_director_roi",
        "writer": "historical_writer_roi"
    }
)

### Merge and Validate Prior Performance Features

The historical genre, director, and writer Reported Return features were merged with the cleaned movie dataset using `imdb_id`. A sample of the resulting records was reviewed to validate the new fields, and the final analysis ready dataset was saved for use in Tableau.


In [24]:
# Add historical genre performance
movies_analysis = movies.merge(
    historical_genre,
    on="imdb_id",
    how="left"
)

# Add historical director and writer performance
movies_analysis = movies_analysis.merge(
    historical_people,
    on="imdb_id",
    how="left"
)

# Confirm the final dataset size
print("Final dataset shape:", movies_analysis.shape)

# Review a sample of films and their historical features
display(
    movies_analysis[
        [
            "title",
            "release_year",
            "genres",
            "ROI",
            "historical_genre_roi",
            "historical_director_roi",
            "historical_writer_roi"
        ]
    ]
    .sort_values("release_year")
    .head(20)
)

# Save the final dataset without modifying the raw data
movies_analysis.to_csv(
    "datasets/cleaned/movies_analysis_final.csv",
    index=False
)

print("movies_analysis_final.csv saved successfully.")

Final dataset shape: (2465, 41)


,title,release_year,genres,ROI,historical_genre_roi,historical_director_roi,historical_writer_roi
0,A Walk Among the Tombstones,2014,"Action,Crime,Drama",1.101228,NaN,NaN,NaN
1,The Giver,2014,"Drama,Romance,Sci-Fi",1.679218,NaN,NaN,NaN
2,The Equalizer,2014,"Action,Crime,Thriller",2.496923,NaN,NaN,NaN
3,Sin City: A Dame to Kill For,2014,"Action,Adventure,Crime",-0.393729,NaN,NaN,NaN
4,Ex Machina,2014,"Drama,Sci-Fi,Thriller",1.457961,NaN,NaN,NaN
5,The Boxtrolls,2014,"Adventure,Animation,Comedy",0.804263,NaN,NaN,NaN
6,Interstellar,2014,"Adventure,Drama,Sci-Fi",3.252904,NaN,NaN,NaN
7,Dracula Untold,2014,"Action,Drama,Fantasy",2.101775,NaN,NaN,NaN
8,Godzilla,2014,"Action,Adventure,Sci-Fi",2.306725,NaN,NaN,NaN
9,Lupin the 3rd,2014,"Action,Adventure,Comedy",1.123810,NaN,NaN,NaN


movies_analysis_final.csv saved successfully.


## Analysis and Findings

### Objective 1: Before Funding Factors and Reported Return
**Method:** Pearson correlation was used to measure the individual linear relationship between reported return and budget, previous lead experience, and historical genre return.

**Result:** Budget recorded a correlation of **-0.044**, previous lead experience **-0.033**, and historical genre return **0.054**.

**Interpretation:** All three values are close to zero, indicating weak individual linear relationships in this selected sample. Correlation results use the observed reported-return values and are sensitive to unusually large films.

### Objective 2: Below Budget Rate by Genre
**Method:** The percentage of films with reported revenue below production budget was calculated for each genre. Films may belong to multiple genres, and the main presentation reports only genres with **at least 25 films**.

**Interpretation:** Genre groups show different below budget rates, but the comparisons are descriptive. Overlapping genre membership and different group sizes mean the chart should not be treated as a funding rule.

### Objective 3: Below Budget Rate by Budget Level
**Method:** Films were grouped into the budget tiers used in the presentation and the below budget rate was compared across tiers.

**Interpretation:** Larger budget groups showed lower below budget rates in the presentation sample. However, budget's overall correlation with reported return remained weak, so higher budget should be treated as context rather than as a rule for funding.

### Objective 4: Previous Lead Experience
**Method:** Films with and without previous lead experience were compared using median reported return and below budget rate.

**Result:** Films with previous lead experience had a **median reported return of 0.73**, **2,303 films**, and a **34.7% below budget rate**. Films without previous lead experience had a **median reported return of 0.05**, **162 films**, and a **35.8% below budget rate**.

**Interpretation:** Median reported return differs substantially, but the below budget rates are nearly unchanged. Previous lead experience therefore shows **no clear standalone signal**.

### Objective 5: Director and Writer Prior History
**Method:** Prior return history uses only earlier year films available inside this selected financial dataset; films released in the same year are not included in the prior history calculation. It is **not full career history**. Films with available history were split at the median prior-history value.

**Coverage:** Director history is available for **633 films**, split into **317 stronger-history** and **316 weaker-history** films. Writer history is available for **792 films**, split into **396 and 396**.

**Result:** Director groups show median current-film reported returns of **1.52 vs 0.67**. Writer groups show **1.91 vs 0.58**.

**Interpretation:** Stronger prior-history groups show higher median reported return descriptively. Because history exists for only part of the sample and is based on available earlier films rather than full careers, this result should be treated as a pattern to validate further rather than a greenlight rule.

### Combined Model
**Method:** OLS regression considers budget, runtime, previous lead experience, genre indicators, prior-history measures, history-availability indicators, and production-company controls together. The reported-return outcome is capped at the **1st and 99th percentile cutoffs**, while films are kept in the model.

**Result:** The expanded model uses **2,442 films**, explains **9.1%** of variation in this sample (**Adjusted R² 7.5%**), and was **not tested on new films**.

**Interpretation:** Runtime shows the clearest relationship in the final robustness interpretation. Budget and the prior-history measures show limited relationships, while previous lead experience does not show a clear standalone signal. The model is exploratory, not a tool for approving or rejecting future films.


## Regression Analysis

The following models document the combined-model analysis. The technical dataset retains the column name `star_power`, but throughout the report this field is interpreted and labeled as **Previous Lead Experience**.

Post-release fields such as IMDb rating, vote count, TMDB popularity, and revenue are not used as predictors. Revenue is used only to construct the reported-return outcome.

The final model is **exploratory**: it was built and checked on the same historical sample and was not externally tested on new films.


In [25]:
import pandas as pd
import statsmodels.api as sm
from collections import Counter
from pathlib import Path

CLEANED_PATH = Path("datasets/cleaned")

# Load the final film dataset used by both the presentation and the models.
# Starting from the saved final file makes the modeling section reproducible
# without depending on variables created in earlier cleaning cells.
df = pd.read_csv(CLEANED_PATH / "movies_analysis_final.csv")

print("Dataset shape:", df.shape)

Dataset shape: (2465, 41)


### Model 1: Core Factors

This baseline model tests budget, runtime, and **Previous Lead Experience** (technical field: `star_power`) against reported return. It is retained to document the progression from a simple specification to the final combined model.


In [26]:
# Model 1 starts with the three core numeric or binary signals used in the project.
# Rows missing one of these fields are removed only for this baseline model so
# statsmodels receives a complete design matrix.
model1_df = df[["ROI", "budget", "runtime_minutes", "star_power"]].dropna()

# A constant is added so the regression estimates an intercept rather than
# forcing the fitted relationship to pass through zero.
X1 = sm.add_constant(
    model1_df[["budget", "runtime_minutes", "star_power"]],
    has_constant="add"
)
y1 = model1_df["ROI"]

model1 = sm.OLS(y1, X1).fit()
print(model1.summary())

                            OLS Regression Results                            
Dep. Variable:                    ROI   R-squared:                       0.003
Model:                            OLS   Adj. R-squared:                  0.002
Method:                 Least Squares   F-statistic:                     2.317
Date:                Wed, 02 Sep 2026   Prob (F-statistic):             0.0738
Time:                        03:55:27   Log-Likelihood:                -10861.
No. Observations:                2442   AIC:                         2.173e+04
Df Residuals:                    2438   BIC:                         2.175e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
const               6.6165      2.552     

### Model 2: Genre Indicators

Genre is categorical, so it is converted to dummy variables. One genre dummy is omitted as the reference category to avoid perfect multicollinearity.


In [27]:
# Genre is stored as comma separated text, while regression requires numeric
# predictors. Each genre is therefore converted into a binary indicator.
genre_dummies = (
    df["genres"]
    .fillna("")
    .str.get_dummies(sep=",")
)

genre_dummies.columns = [
    "genre_" + c.strip().replace(" ", "_")
    for c in genre_dummies.columns
]

# One genre indicator is omitted as the reference category. Keeping every
# indicator together with an intercept would create perfect redundancy.
if genre_dummies.shape[1] > 1:
    genre_dummies = genre_dummies.iloc[:, 1:]

X2 = pd.concat(
    [df[["budget", "runtime_minutes", "star_power"]], genre_dummies],
    axis=1
).apply(pd.to_numeric, errors="coerce")

model2_data = pd.concat([df[["ROI"]], X2], axis=1).dropna()

y2 = model2_data["ROI"]
X2_clean = sm.add_constant(model2_data.drop(columns="ROI"), has_constant="add")

model2 = sm.OLS(y2, X2_clean).fit()
print(model2.summary())

                            OLS Regression Results                            
Dep. Variable:                    ROI   R-squared:                       0.018
Model:                            OLS   Adj. R-squared:                  0.008
Method:                 Least Squares   F-statistic:                     1.804
Date:                Wed, 02 Sep 2026   Prob (F-statistic):            0.00966
Time:                        03:55:27   Log-Likelihood:                -10842.
No. Observations:                2442   AIC:                         2.173e+04
Df Residuals:                    2417   BIC:                         2.188e+04
Df Model:                          24                                         
Covariance Type:            nonrobust                                         
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                 5.8638      3.23

### Model 3: Prior Performance History

Historical genre, director, and writer return measures use **earlier release years only**. Director and writer history represents prior history available inside this selected financial dataset, **does not represent complete careers**.

Missing historical values are not treated as observed zero history without context: each value is filled with zero for modeling and paired with an availability indicator showing whether history was originally available.


In [28]:
historical_cols = [
    "historical_genre_roi",
    "historical_director_roi",
    "historical_writer_roi"
]

model_base = df.copy()

# A missing historical ROI usually means that no earlier film history was
# available for that genre or person. Replacing it with zero keeps the film in
# the model, while a separate availability flag tells the model that the zero
# represents missing history rather than an observed historical ROI of zero.
for col in historical_cols:
    model_base[col + "_available"] = model_base[col].notna().astype(int)
    model_base[col] = model_base[col].fillna(0)

core_cols = [
    "budget",
    "runtime_minutes",
    "star_power",
    "historical_genre_roi",
    "historical_director_roi",
    "historical_writer_roi",
    "historical_genre_roi_available",
    "historical_director_roi_available",
    "historical_writer_roi_available"
]

# Combine the core factors, historical measures, availability flags, and genre
# indicators so Model 3 tests them together rather than one at a time.
X3 = pd.concat(
    [model_base[core_cols], genre_dummies],
    axis=1
).apply(pd.to_numeric, errors="coerce")

model3_data = pd.concat([model_base[["ROI"]], X3], axis=1).dropna()

y3 = model3_data["ROI"]
X3_clean = sm.add_constant(model3_data.drop(columns="ROI"), has_constant="add")

model3 = sm.OLS(y3, X3_clean).fit()
print(model3.summary())

                            OLS Regression Results                            
Dep. Variable:                    ROI   R-squared:                       0.022
Model:                            OLS   Adj. R-squared:                  0.010
Method:                 Least Squares   F-statistic:                     1.787
Date:                Wed, 02 Sep 2026   Prob (F-statistic):            0.00546
Time:                        03:55:27   Log-Likelihood:                -10837.
No. Observations:                2442   AIC:                         2.174e+04
Df Residuals:                    2411   BIC:                         2.192e+04
Df Model:                          30                                         
Covariance Type:            nonrobust                                         
                                        coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------------------------
const 

### Model 4: Combined Model with Capped Reported Return

Film reported return contains unusually large values. For the final base regression, the **outcome is capped at the 1st and 99th percentile cutoffs**. The films themselves are retained.

This treatment is used for the regression only. Descriptive group charts use medians of the observed current-film reported-return values unless otherwise stated.


In [29]:
# ROI contains a small number of extreme values. For the final regression,
# only the target is capped at the 1st and 99th percentiles. This reduces the
# influence of extreme returns without changing the descriptive analyses,
# which continue to use the observed ROI values.
roi_low = model_base["ROI"].quantile(0.01)
roi_high = model_base["ROI"].quantile(0.99)
model_base["ROI_winsorized"] = model_base["ROI"].clip(roi_low, roi_high)

# Model 4 keeps the same predictors as Model 3. The main change is the capped
# ROI target, allowing a fair comparison with the later company control model.
X4 = pd.concat(
    [model_base[core_cols], genre_dummies],
    axis=1
).apply(pd.to_numeric, errors="coerce")

model4_data = pd.concat(
    [model_base[["ROI_winsorized"]], X4],
    axis=1
).dropna()

y4 = model4_data["ROI_winsorized"]
X4_clean = sm.add_constant(
    model4_data.drop(columns="ROI_winsorized"),
    has_constant="add"
)

model4 = sm.OLS(y4, X4_clean).fit()

print("N:", int(model4.nobs))
print("R²:", round(model4.rsquared, 4))
print("Adjusted R²:", round(model4.rsquared_adj, 4))
print("Overall p value:", model4.f_pvalue)
print(model4.summary())

N: 2442
R²: 0.047
Adjusted R²: 0.0352
Overall p value: 3.475857790687422e-12
                            OLS Regression Results                            
Dep. Variable:         ROI_winsorized   R-squared:                       0.047
Model:                            OLS   Adj. R-squared:                  0.035
Method:                 Least Squares   F-statistic:                     3.966
Date:                Wed, 02 Sep 2026   Prob (F-statistic):           3.48e-12
Time:                        03:55:27   Log-Likelihood:                -7617.0
No. Observations:                2442   AIC:                         1.530e+04
Df Residuals:                    2411   BIC:                         1.548e+04
Df Model:                          30                                         
Covariance Type:            nonrobust                                         
                                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------

**Model 4 comparison result:** N = **2,442**; R² = **0.0470**; Adjusted R² = **0.0352**. Runtime shows the clearest relationship in this specification; the other main factors show limited independent relationships.


### Model 5: Production Company Context

The expanded model used in the presentation adds production-company context to Model 4. Production companies are known before release, but they are treated as **context/control variables**, not approval rules.

Only companies appearing in at least 50 films are included to limit sparse indicators. Films may list more than one production company, so these controls can overlap.


In [30]:
# A film can list several production companies in one text field. Split the
# field into a list so each company can be tested as its own binary control.
model_base["company_list"] = (
    model_base["production_companies"]
    .fillna("")
    .apply(lambda x: [c.strip() for c in x.split(",") if c.strip()])
)

# Count each company once per film. Using set(companies) prevents accidental
# duplicate company names within one film from increasing the count.
company_counts = Counter(
    company
    for companies in model_base["company_list"]
    for company in set(companies)
)

# Keep companies appearing in at least 50 films. This limits sparse controls
# that would add many parameters with very little supporting data.
major_companies = sorted([
    company
    for company, count in company_counts.items()
    if count >= 50
])

company_dummies = pd.DataFrame(
    {
        "company_" + company: model_base["company_list"].apply(
            lambda companies: int(company in companies)
        )
        for company in major_companies
    },
    index=model_base.index
)

print("Production company controls:", len(major_companies))
print(major_companies)

Production company controls: 11
['20th Century Fox', 'Blumhouse Productions', 'Canal+', 'Columbia Pictures', 'Lionsgate', 'New Line Cinema', 'Paramount', 'TSG Entertainment', 'Universal Pictures', 'Walt Disney Pictures', 'Warner Bros. Pictures']


In [31]:
# Add production company controls to the exact predictor set used in Model 4.
# No original factor is removed, so the change in model fit can be attributed
# to the added production company context.
X5 = pd.concat(
    [X4, company_dummies],
    axis=1
).apply(pd.to_numeric, errors="coerce")

# Use exactly the same film rows as Model 4. Holding the sample constant makes
# the R squared comparison fair and avoids improvement caused by sample changes.
model5_data = pd.concat(
    [model_base[["ROI_winsorized"]], X5],
    axis=1
).loc[model4_data.index].dropna()

y5 = model5_data["ROI_winsorized"]
X5_clean = sm.add_constant(
    model5_data.drop(columns="ROI_winsorized"),
    has_constant="add"
)

model5 = sm.OLS(y5, X5_clean).fit()

print("BASE MODEL")
print("N:", int(model4.nobs))
print("R²:", round(model4.rsquared, 4))
print("Adjusted R²:", round(model4.rsquared_adj, 4))

print("\nWITH PRODUCTION COMPANY CONTROL")
print("N:", int(model5.nobs))
print("R²:", round(model5.rsquared, 4))
print("Adjusted R²:", round(model5.rsquared_adj, 4))
print("Overall p value:", model5.f_pvalue)

print("\nCHANGE")
print("R² change:", round(model5.rsquared - model4.rsquared, 4))
print("Adjusted R² change:", round(model5.rsquared_adj - model4.rsquared_adj, 4))

BASE MODEL
N: 2442
R²: 0.047
Adjusted R²: 0.0352

WITH PRODUCTION COMPANY CONTROL
N: 2442
R²: 0.0906
Adjusted R²: 0.0751
Overall p value: 4.255764843244119e-28

CHANGE
R² change: 0.0436
Adjusted R² change: 0.0399


**Expanded model result used in the presentation:** N = **2,442**; R² = **0.0906 (9.1%)**; Adjusted R² = **0.0751 (7.5%)**.

The model explains only a limited share of differences in reported return within this historical sample and was **not tested on new films**. It should therefore be interpreted as exploratory rather than predictive.


### Robustness Check

Ordinary OLS standard errors can be sensitive to unequal error variance, which is plausible in film-return data. The final interpretation therefore uses a **heteroskedasticity-robust (HC3) error check** in addition to the original OLS specification.

The purpose of this check is not to turn p-values into a greenlight rule. It is to avoid presenting a factor as a clear standalone signal when its inference changes under a reasonable alternative error specification.


In [32]:
# HC3 robustness check for the same expanded Model 5 specification.
# Coefficients and model fit are unchanged; HC3 changes the estimated
# uncertainty around the coefficients.

model5_hc3 = model5.get_robustcov_results(cov_type="HC3")

hc3_params = pd.Series(model5_hc3.params, index=model5.params.index)
hc3_pvalues = pd.Series(model5_hc3.pvalues, index=model5.params.index)

main_factors = [
    "budget",
    "runtime_minutes",
    "star_power",
    "historical_genre_roi",
    "historical_director_roi",
    "historical_writer_roi"
]

factor_results = pd.DataFrame({
    "Coefficient": model5.params[main_factors],
    "OLS_P_Value": model5.pvalues[main_factors],
    "HC3_P_Value": hc3_pvalues[main_factors]
})

factor_results["Presentation_Label"] = [
    "Limited relationship",
    "Clear relationship",
    "No clear standalone signal",
    "Limited relationship",
    "Limited relationship",
    "Limited relationship"
]

print(factor_results.round(4))


                         Coefficient  OLS_P_Value  HC3_P_Value  \
budget                       -0.0000       0.0886       0.0349   
runtime_minutes               0.0224       0.0001       0.0007   
star_power                   -1.1298       0.0342       0.1578   
historical_genre_roi         -0.1010       0.3731       0.5696   
historical_director_roi      -0.0063       0.5223       0.3195   
historical_writer_roi         0.0026       0.8258       0.8256   

                                 Presentation_Label  
budget                         Limited relationship  
runtime_minutes                  Clear relationship  
star_power               No clear standalone signal  
historical_genre_roi           Limited relationship  
historical_director_roi        Limited relationship  
historical_writer_roi          Limited relationship  


Using the stronger HC3 error check, **runtime** remains the clearest factor. **Previous Lead Experience** does not provide a clear standalone signal under the robustness check, even though the ordinary OLS p-value was smaller in one specification. Budget and the historical genre, director, and writer measures remain limited.

This is the interpretation used in the final presentation. The model does not establish causation and was not tested on new films.


## Production Company Context

Production company indicators are included only as contextual controls in the expanded regression. They are not used as greenlight rules or as recommendations in the final presentation.

The following checks are retained only to document how company context was explored during model development.


In [33]:
# The significant company controls are checked descriptively to make sure the
# regression result is not supported only by one extreme reported return observation.
# Median reported return is resistant to outliers, while the winsorized mean matches the
# treatment used for the final regression target.
validation = []

for company in ["Blumhouse Productions", "New Line Cinema"]:
    mask = model_base["company_list"].apply(lambda companies: company in companies)
    temp = model_base.loc[mask].copy()

    validation.append({
        "Company": company,
        "Films": len(temp),
        "Median_Reported_Return": temp["ROI"].median(),
        "Capped_Mean_Reported_Return": temp["ROI_winsorized"].mean(),
        "Below_Budget_Rate_%": (temp["ROI"] < 0).mean() * 100
    })

validation = pd.DataFrame(validation)
print(validation.round(2).to_string(index=False))

              Company  Films  Median_Reported_Return  Capped_Mean_Reported_Return  Below_Budget_Rate_%
Blumhouse Productions     53                    9.35                        10.86                11.32
      New Line Cinema     50                    2.70                         5.74                 8.00


Validation results: Blumhouse Productions has 53 films, median Reported Return 9.35, winsorized mean Reported Return 10.86, and 11.3% below budget rate. New Line Cinema has 50 films, median Reported Return 2.70, winsorized mean Reported Return 5.74, and 8.0% below budget rate. These results support using production company as contextual evidence. The overlapping company structure and non causal design remain limitations.


## SQL Queries for Production Context

The following queries were run in DB Browser for SQLite during the final model review. They are documented here so the extraction can be reproduced. The production country export was exploratory. Production country was **not retained in the final presentation model** because production company provided a clearer business context control.


### Review Available Columns
```sql
SELECT
    'IMDb title_basics' AS source,
    GROUP_CONCAT(name, ', ') AS columns
FROM pragma_table_info('title_basics')
UNION ALL
SELECT
    'TMDB' AS source,
    GROUP_CONCAT(name, ', ') AS columns
FROM pragma_table_info('tmdb');
```


### Export Production Context Fields
```sql
SELECT
    imdb_id,
    production_companies,
    production_countries
FROM tmdb
WHERE imdb_id IS NOT NULL
  AND imdb_id != ''
  AND budget > 0
  AND revenue > 0;
```

The exported file was used only to test whether production country context added explanatory power. `production_companies` was already present in the final main cleaned dataset, so no extra production company file is required for the final analysis.


## Final Dataset Lineage

### Final cleaned files
1. **`movies_analysis_final.csv`**, final film level analytical dataset: **2,465 rows, 41 columns**, 2014 to 2023.
2. **`cast_crew_filtered.csv`**, cast/crew linkage table: **37,483 rows, 6 columns**.

### Reproducibility Funnel
**227,265 IMDb title records after the initial filter + 486,063 TMDB rows → 102,868 matched rows → 2,826 rows after usable reported budget/revenue and below-$1,000 exclusions → 2,465 unique 2014 to 2023 IMDb films.**

**Matching rule:** IMDb `tconst` → TMDB `imdb_id`.

**Final duplicate check:** 0 duplicate `imdb_id` rows.

**Financial rule:** reported budget and reported revenue must both be usable and at least **$1,000**.

### Important Coverage Checks
- Missing runtime: **23 films**.
- Director prior-return history: **633 / 2,465 films (25.7%)**.
- Writer prior-return history: **792 / 2,465 films (32.1%)**.
- Prior director/writer return history uses earlier year films available in this selected dataset and is **not full career history**.
- Same-year releases are excluded from prior-history calculations.

### Lineage Summary
Raw IMDb + TMDB → SQLite/filtering → `tconst`/`imdb_id` match → usable financial-value rules → unique 2014 to 2023 film sample → cast/crew linkage → previous-experience flags → prior-history features → descriptive analysis and combined regression.


## Statistical Checks

Formal tests are kept in the technical report as supporting checks rather than as the main presentation story.

Because reported return is strongly skewed and contains unusually large values:
- descriptive group comparisons emphasize **median reported return**;
- the combined regression caps the **outcome** at the 1st and 99th percentile cutoffs while keeping films;
- director/writer prior-history sensitivity is recalculated using capped earlier-film return values;
- the expanded regression is also checked with **HC3 robust standard errors**.

A p-value does not prove practical importance, causation, or predictive usefulness.


In [34]:
from scipy import stats
from pathlib import Path
import pandas as pd

# Load the final analysis file if this section is run independently.
if "df" not in globals():
    cleaned_candidates = [
        Path("datasets/cleaned/movies_analysis_final.csv"),
        Path("datasets/cleaned/movies_analysis_final(7).csv"),
        Path("movies_analysis_final.csv"),
        Path("movies_analysis_final(7).csv")
    ]
    analysis_file = next((p for p in cleaned_candidates if p.exists()), None)
    if analysis_file is None:
        raise FileNotFoundError(
            "Final analysis CSV not found. Place movies_analysis_final.csv in datasets/cleaned/."
        )
    df = pd.read_csv(analysis_file)

significance_results = []

# Objective 2: Genre
genre_cols = [c for c in df.columns if c.startswith("genre_")]
genre_groups = [
    df.loc[df[c] == 1, "ROI"].dropna()
    for c in genre_cols
    if (df[c] == 1).sum() >= 25
]

h_genre, p_genre = stats.kruskal(*genre_groups)
significance_results.append(
    ("Genre (Objective 2)", "Kruskal-Wallis", h_genre, p_genre)
)

# Objective 3: Budget level
bins = [0, 5e6, 10e6, 25e6, 50e6, 100e6, float("inf")]
tier_labels = [
    "Under $5M",
    "$5M to $10M",
    "$10M to $25M",
    "$25M to $50M",
    "$50M to $100M",
    "$100M+"
]

budget_tier = pd.cut(
    df["budget"],
    bins=bins,
    labels=tier_labels,
    right=False
)

budget_groups = [
    df.loc[budget_tier == label, "ROI"].dropna()
    for label in tier_labels
]

h_budget, p_budget = stats.kruskal(*budget_groups)
significance_results.append(
    ("Budget level (Objective 3)", "Kruskal-Wallis", h_budget, p_budget)
)

# Objective 4: Previous lead experience
lead_experience = df.loc[df["star_power"] == 1, "ROI"].dropna()
no_lead_experience = df.loc[df["star_power"] == 0, "ROI"].dropna()

u_lead, p_lead = stats.mannwhitneyu(
    lead_experience,
    no_lead_experience,
    alternative="two-sided"
)

significance_results.append(
    (
        "Previous lead experience (Objective 4)",
        "Mann-Whitney U",
        u_lead,
        p_lead
    )
)

# Objective 5: Director and writer prior history
for col, role in [
    ("historical_director_roi", "Director"),
    ("historical_writer_roi", "Writer")
]:
    role_df = df[df[col].notna()]
    role_median = role_df[col].median()

    higher = role_df.loc[role_df[col] >= role_median, "ROI"].dropna()
    lower = role_df.loc[role_df[col] < role_median, "ROI"].dropna()

    u_role, p_role = stats.mannwhitneyu(
        higher,
        lower,
        alternative="two-sided"
    )

    significance_results.append(
        (
            f"{role} prior history (Objective 5)",
            "Mann-Whitney U",
            u_role,
            p_role
        )
    )

significance_table = pd.DataFrame(
    significance_results,
    columns=["Comparison", "Test", "Statistic", "P_Value"]
)

significance_table["Significant (p < 0.05)"] = (
    significance_table["P_Value"] < 0.05
)

print(significance_table.round(4).to_string(index=False))


                            Comparison           Test   Statistic  P_Value  Significant (p < 0.05)
                   Genre (Objective 2) Kruskal-Wallis    113.7279   0.0000                    True
            Budget level (Objective 3) Kruskal-Wallis     59.7599   0.0000                    True
Previous lead experience (Objective 4) Mann-Whitney U 199616.5000   0.1354                   False
  Director prior history (Objective 5) Mann-Whitney U  62361.5000   0.0000                    True
    Writer prior history (Objective 5) Mann-Whitney U 102785.0000   0.0000                    True


**Interpretation:** Formal group tests are secondary to the descriptive findings and should not be read as proof that a factor should be used for approval decisions.

Previous lead experience is especially important to interpret cautiously: the descriptive median reported return differs between the two groups, but below budget rates are nearly identical, the standalone group test is not statistically significant, and the stronger regression error check does not support a clear standalone signal.

The director/writer history pattern is also treated cautiously because the history fields cover only part of the sample and are not full career histories.


## Final Verification

This section is designed to run directly from the submitted final CSV files. It verifies the key sample facts requested in the mock feedback and checks whether the creative team prior-history conclusion changes when unusually large **earlier-film** reported returns are capped at the 1st and 99th percentile cutoffs.

The cap is applied when constructing the historical director/writer averages, not to the current-film median reported return shown in the presentation.


In [35]:
from pathlib import Path
import pandas as pd
import numpy as np

# Locate submitted files either beside the notebook or in datasets/cleaned.
candidates = [Path("."), Path("datasets/cleaned"), Path("/mnt/data")]
base = next(
    p for p in candidates
    if (p / "movies_analysis_final(7).csv").exists()
    or (p / "movies_analysis_final.csv").exists()
)

movie_file = (
    base / "movies_analysis_final(7).csv"
    if (base / "movies_analysis_final(7).csv").exists()
    else base / "movies_analysis_final.csv"
)
crew_file = (
    base / "cast_crew_filtered(7).csv"
    if (base / "cast_crew_filtered(7).csv").exists()
    else base / "cast_crew_filtered.csv"
)

movies_check = pd.read_csv(movie_file)
crew_check = pd.read_csv(crew_file)

print("FINAL SAMPLE CHECK")
print("Films:", len(movies_check))
print("Years:", int(movies_check["release_year"].min()), "-", int(movies_check["release_year"].max()))
print("Duplicate imdb_id:", int(movies_check["imdb_id"].duplicated().sum()))
print("Missing runtime:", int(movies_check["runtime_minutes"].isna().sum()))
print("Below-budget films:", int((movies_check["ROI"] < 0).sum()))
print("Below-budget rate:", round((movies_check["ROI"] < 0).mean() * 100, 1), "%")
shortfall = (movies_check.loc[movies_check["ROI"] < 0, "budget"]
             - movies_check.loc[movies_check["ROI"] < 0, "revenue"]).mean()
print("Average shortfall among below-budget films: $", round(shortfall / 1e6, 2), "M")
print("Director history coverage:", int(movies_check["historical_director_roi"].notna().sum()))
print("Writer history coverage:", int(movies_check["historical_writer_roi"].notna().sum()))

# Cap earlier-film ROI at the same 1st/99th percentile cutoffs used as the
# regression sensitivity treatment, then rebuild director/writer history.
lo, hi = movies_check["ROI"].quantile([0.01, 0.99])
movies_check["ROI_capped_for_history"] = movies_check["ROI"].clip(lo, hi)

print("\nHISTORY SENSITIVITY CUT-OFFS")
print("1st percentile:", round(lo, 4))
print("99th percentile:", round(hi, 4))

def rebuild_capped_history(role):
    credits = (
        crew_check.loc[crew_check["category"].eq(role), ["imdb_id", "nconst"]]
        .drop_duplicates()
    )
    x = credits.merge(
        movies_check[["imdb_id", "release_year", "ROI_capped_for_history"]],
        on="imdb_id",
        how="inner"
    )

    person_year = (
        x.groupby(["nconst", "release_year"], as_index=False)
        .agg(
            year_roi_sum=("ROI_capped_for_history", "sum"),
            year_n=("ROI_capped_for_history", "size")
        )
        .sort_values(["nconst", "release_year"])
    )

    person_year["prior_sum"] = (
        person_year.groupby("nconst")["year_roi_sum"].cumsum()
        - person_year["year_roi_sum"]
    )
    person_year["prior_n"] = (
        person_year.groupby("nconst")["year_n"].cumsum()
        - person_year["year_n"]
    )
    person_year["history_capped"] = (
        person_year["prior_sum"] / person_year["prior_n"].replace(0, np.nan)
    )

    film_history = (
        x[["imdb_id", "nconst", "release_year"]]
        .drop_duplicates()
        .merge(
            person_year[["nconst", "release_year", "history_capped"]],
            on=["nconst", "release_year"],
            how="left"
        )
        .groupby("imdb_id", as_index=False)["history_capped"]
        .mean()
    )

    result = (
        movies_check[["imdb_id", "ROI"]]
        .merge(film_history, on="imdb_id", how="left")
        .dropna(subset=["history_capped"])
    )

    split = result["history_capped"].median()
    result["history_group"] = np.where(
        result["history_capped"] >= split,
        "Stronger history",
        "Weaker history"
    )

    summary = (
        result.groupby("history_group")
        .agg(
            films=("ROI", "size"),
            median_current_reported_return=("ROI", "median"),
            below_budget_rate=("ROI", lambda s: (s < 0).mean() * 100)
        )
        .round(3)
    )
    return result, summary

for role in ["director", "writer"]:
    role_data, role_summary = rebuild_capped_history(role)
    print(f"\n{role.upper()} HISTORY AFTER CAPPING EARLIER-FILM RETURNS")
    print("Coverage:", len(role_data))
    print(role_summary)


FINAL SAMPLE CHECK
Films: 2465
Years: 2014 - 2023
Duplicate imdb_id: 0
Missing runtime: 23
Below-budget films: 858
Below-budget rate: 34.8 %
Average shortfall among below-budget films: $ 11.05 M
Director history coverage: 633
Writer history coverage: 792

HISTORY SENSITIVITY CUT-OFFS
1st percentile: -0.9972
99th percentile: 38.7093

DIRECTOR HISTORY AFTER CAPPING EARLIER-FILM RETURNS
Coverage: 633
                  films  median_current_reported_return  below_budget_rate
history_group                                                             
Stronger history    317                           1.521             18.927
Weaker history      316                           0.672             37.025

WRITER HISTORY AFTER CAPPING EARLIER-FILM RETURNS
Coverage: 792
                  films  median_current_reported_return  below_budget_rate
history_group                                                             
Stronger history    396                           1.915             17.929
Weaker hi

### Verification Result

The sensitivity check leaves the presentation's creative team pattern unchanged:

- **Directors:** 633 films; stronger-history group **317 films, median reported return 1.52** vs weaker-history group **316 films, 0.67**.
- **Writers:** 792 films; stronger-history group **396 films, median reported return 1.91** vs weaker-history group **396 films, 0.58**.

This supports keeping the descriptive Creative Team slide, with the stated limitation that prior history is available only for part of the sample and is not full career history.


## Recommendations

1. **Combine Signals Before Greenlighting**  
   Consider multiple factors rather than relying on one alone.

2. **Compare Genre Funding Outcomes**  
   Consider below budget rates when comparing genre options.

3. **Treat Budget as Context**  
   Compare budget levels without treating higher budgets as a funding rule.

4. **Don't Use Lead Experience Alone**  
   Previous lead experience showed no clear standalone signal.

5. **Further Validate Creative Team History**  
   Use broader career data before applying prior history to decisions.


## Limitations

### Primary Limitations

1. **Simplified Reported Return Measure**  
   Reported return compares reported revenue with production budget only. It excludes marketing, distribution, cinema/distributor shares, financing costs, and investor payments, so it is **not actual investor profit or true break-even**.

2. **Selected Financial Sample**  
   The analysis contains selected 2014 to 2023 films with usable reported budget and revenue data, not all films.

3. **Extreme Return Values**  
   Unusually large reported-return values can affect averages and rankings. Descriptive charts therefore emphasize medians; the regression caps the outcome at the 1st and 99th percentile cutoffs while keeping films.

4. **Limited Creative-Team History**  
   Director history is available for 633 films and writer history for 792 films. These fields use earlier year films available in this selected dataset and **do not represent complete careers**.

5. **Model Not Externally Tested**  
   The final model was built and checked on the same **2,442 films** and explains a limited share of variation. It was not validated on unseen films and is **not a prediction or approval tool**.

### Secondary Technical Notes
- Films can belong to multiple genres, so genre groups overlap.
- Films can list multiple production companies, so company indicators overlap.
- 23 films have missing runtime and are excluded from runtime-specific comparisons/model rows requiring runtime.
- Budget and revenue are treated as nominal reported USD values and were not adjusted for inflation.
- Budget/revenue values below $1,000 were excluded as a project data quality rule.
- Same-year releases are excluded from prior-history calculations.
- Correlations, group comparisons, and regression coefficients describe relationships and do not establish causation.


## Assumptions

1. Reported budget and revenue values are treated as comparable reported financial figures for this exploratory analysis.
2. The reported return measure compares reported revenue with production budget only and does not represent actual investor profit or break even.
3. The $1,000 minimum for reported budget and revenue is a project data quality rule used to exclude implausible or unusable financial entries.
4. Films may belong to more than one genre, so genre groups are not mutually exclusive.
5. Previous lead experience indicates whether the selected lead had earlier qualifying credits in the available data. It is not a measure of celebrity or star power.
6. Director and writer prior history uses earlier year films available in the selected financial dataset. Same year releases are excluded, and the history does not represent complete careers.
7. The regression is exploratory. It was built and checked on the same 2,442 films and was not tested on new films.


## References

1. **IMDb.** *IMDb Non-Commercial Datasets.* `https://datasets.imdbws.com/`, Accessed August 2026.
2. **asaniczka.** *Full TMDB Movies Dataset 2024 (1M Movies).* Kaggle. Source file used: `TMDB_movie_dataset_v11.csv`. `https://www.kaggle.com/datasets/asaniczka/tmdb-movies-dataset-2023-930k-movies`, Accessed August 2026.
3. **Deniz, B., & Hasbrouck, R. B.** (2012). *When to Greenlight: Examining the Pre-Release Factors That Determine Future Box Office Success of a Movie in the United States.* International Journal of Economics and Management Sciences, 2(3), 35–42. `https://www.hilarispublisher.com/open-access/when-to-greenlight-examining-the-prerelease-factors-that-determine-future-box-office-success-of-a-movie-in-the-united-states-2162-6359-2-127.pdf`, Accessed August 2026.


## AI Usage Disclosure

AI tools were used throughout the project development process for brainstorming, code debugging, analytical logic review, documentation, and presentation wording. They were used to support the work rather than as a source of project evidence. Final calculations, code outputs, analytical decisions, and interpretations were checked against the submitted project data before inclusion. The author remains responsible for the final analysis and presentation choices.
